scene/2d/tile_map_layer.cpp

```diff
@@ -1623,32 +1687,29 @@ void TileMapLayer::_notification(int p_what) {
 		case NOTIFICATION_ENTER_TREE: {
 			_update_notify_local_transform();
 			dirty.flags[DIRTY_FLAGS_LAYER_IN_TREE] = true;
-			TileMap *tile_map_node = _fetch_tilemap();
-			tile_map_node->queue_internal_update();
+			_queue_internal_update();
 		} break;
 
 		case NOTIFICATION_EXIT_TREE: {
 			dirty.flags[DIRTY_FLAGS_LAYER_IN_TREE] = true;
-			TileMap *tile_map_node = _fetch_tilemap();
-			tile_map_node->queue_internal_update();
+			// Update immediately on exiting, alternatively we can simply skip this update and let it be done on entering or on request.
+			update_internals();
```

Comment:
This part is a point where I'm unsure about the exact solution

Reply:
No that is the right solution. You need to trigger an update when leaving the tree to free the CanvasItems, physics shapes, etc... anyway. Users would complain if removing a TileMap from the tree would still have CanvasItems or collisions shapes visible/colliding.

Reply:
Ah and we probably want to remove the comment here. At least the "alternatively part" I guess :sweat_smile:

Reply:
Already done :)